metricflow/dataflow/optimizer/predicate_pushdown_optimizer.py

```diff
@@ -233,10 +233,17 @@ def _push_down_where_filters(
 
         for filter_spec in current_pushdown_state.where_filter_specs:
             filter_spec_semantic_models = self._models_for_spec(filter_spec)
+            invalid_element_types = [
+                element
+                for element in filter_spec.linkable_elements
+                if element.element_type not in current_pushdown_state.pushdown_eligible_element_types
+            ]
+            if len(filter_spec_semantic_models) != 1 or len(invalid_element_types) > 0:
+                continue
+
             all_linkable_specs_match = all(spec in source_node_linkable_specs for spec in filter_spec.linkable_specs)
-            semantic_models_match = (
-                len(filter_spec_semantic_models) == 1 and filter_spec_semantic_models[0] == source_semantic_model
-            )
+            # TODO: Handle the case where entities can be defined in multiple models, only one of which need match
```

Comment:
Is there a task up for this / is it handled up the stack? Just want to make sure it doesn't get forgotten.

Reply:
This will be handled once entities are enabled. I'm using #1011 for all of that so I'll make some notes in there!